includes/functions.php

```diff
@@ -452,34 +452,37 @@ function bpeo_remove_default_canonical_event_content( $retval ) {
  * @see bpeo_remove_default_canonical_event_content()
  *
  * @param  string $content Current content.
- * @return string
+ * @return string $content The modified content.
  */
 function bpeo_canonical_event_content( $content ) {
 	global $pages;
 
-	// reset get_the_content() to use already-rendered content so we can use it in
-	// our content-eo-event.php template part
-	//
-	// get_the_content() is weird and checks the $pages global for the content
-	// so let's use the rendered content here and set it in the $pages global
-	$pages[0] = $content;
+	// sanity check
+	if ( is_singular( 'event' ) && get_post_type( get_the_ID() ) == 'event' ) {
```

Comment:
Can we do the reverse check and return the `$content` variable here if it doesn't match the condition?

Would rather do that than doing a set of indents.

Reply:
@r-a-y I left the conditional that way because I was unsure whether `remove_filter( 'eventorganiser_template_stack', 'bpeo_register_template_stack' );` needed to be called regardless of whether the conditions were met. In the end, I nested that line inside the conditional, so as it stands, I take your point :)

Would you prefer me to add a commit with reverse checks, or close this and raise a new PR?